react/features/lobby/middleware.ts

```diff
@@ -315,17 +323,23 @@ function _conferenceFailed({ dispatch, getState }: IStore, next: Function, actio
         return result;
     }
 
-    dispatch(hideLobbyScreen());
+    // if both are available pre-join is with priority (the case when pre-join is enabled)
+    if (isPrejoinPageVisible(state)) {
+        dispatch(hideLobbyScreen());
+    }
 
     if (error.name === JitsiConferenceErrors.CONFERENCE_ACCESS_DENIED) {
         dispatch(
             showNotification({
                 appearance: NOTIFICATION_TYPE.ERROR,
                 hideErrorSupportLink: true,
                 titleKey: 'lobby.joinRejectedTitle',
+                uid: LOBBY_NOTIFICATION_ID,
                 descriptionKey: 'lobby.joinRejectedMessage'
-            }, NOTIFICATION_TIMEOUT_TYPE.LONG)
+            }, NOTIFICATION_TIMEOUT_TYPE.STICKY)
         );
+    } else {
+        dispatch(hideLobbyScreen());
```

Comment:
This looks a little bit strange. I think in some cases we will dispatch `hideLobbyScreen` 2 times. Like if the error is not access denied and there is prejoin.

I'm not totally sure what is the intent here. When do we actually want to hide the lobby screen?